### PR TITLE
Remove more env vars

### PR DESCRIPTION
**Pull request by eessex**
Removes logic related to 
- `MAILCHIMP_KEY`
- `MOBILE_URL`
- `GALLERY_INSIGHTS_LIST`
- `CANONICAL_MOBILE_URL`

### CODE REVIEW

File: src/desktop/components/article/client/view.coffee [hunk 1]
```diff
@@ -42,8 +42,6 @@ module.exports = class ArticleView extends Backbone.View
     @setupFollowButtons()
     @setupImageSets()
     @resetImageSetPreview()
-    if @article.attributes.channel?.id == sd.GALLERY_INSIGHTS_CHANNEL
-      @setupMarketoStyles()
```
Comment (joeyAghion) [inline src/desktop/components/article/client/view.coffee]: I see some other references to `GALLERY_INSIGHTS_CHANNEL` in the codebase. Can those be removed as well?

**Comment by joeyAghion** [comment]
This is mergeable as far as I'm concerned, but it might be worth deploying the earlier clean-up (https://github.com/artsy/force/pull/4528) first. It just depends how safe you think each step is.

File: src/desktop/components/article/client/view.coffee (hunk 1 above)
Comment (eessex) [reply src/desktop/components/article/client/view.coffee]: This var is still used to redirect GI insights articles to `partners.artsy.net`

**Comment by eessex** [comment]
I think we are ok to merge this now, can double check staging for the moment until force deploys are unblocked.

**Pull request merged**